lib/shared/bucketing-assembly-script/assembly/bucketing/segmentation.ts

```diff
@@ -2,10 +2,21 @@ import { RegExp } from 'assemblyscript-regex/assembly'
 import {  findString, includes, replace} from '../helpers/lodashHelpers'
 import { OptionsType, versionCompare } from './versionCompare'
 import {
-    TopLevelOperator, AudienceFilterOrOperator, DVCPopulatedUser, validSubTypes
+    TopLevelOperator, AudienceFilterOrOperator, DVCPopulatedUser, validSubTypes,
+    DVCUser, Target as PublicTarget
 } from '../types'
 import { JSON } from 'assemblyscript-json/assembly'
-import { getF64FromJSONValue } from '../helpers/jsonHelpers'
+import {getF64FromJSONObj, getF64FromJSONValue} from '../helpers/jsonHelpers'
+
+export function evaluateOperatorFromJSON(operatorStr: string, userStr: string): bool {
```

Comment:
maybe it would be cleaner if you collected all the public interface methods in one file and then just had them call the methods from the existing bucketing and segmentation files? With the current structure it's pretty hard to easily see what the interface of this library is.